File: demo-task-executor-java/src/main/java/demo/logger/ConsoleLogger.java
package demo.logger;

/**
 * Logger that writes messages to the console ("standard" output stream).
 * This class IS thread-safe.
 */
public class ConsoleLogger extends AbstractLogger {

    @Override
    synchronized public void log(Severity severity, Object message) {
        System.out.println(getMessagePrefix(severity) + message);
    }
}

File: demo-task-executor-scala/src/main/scala/demo/logger/ConsoleLogger.scala
package demo.logger

/**
  * Logger that writes messages to the console ("standard" output stream).
  * This class IS thread-safe.
  */
class ConsoleLogger extends AbstractLogger {

    override def log(severity: Severity, message: Any) {
        this.synchronized {
            println(getMessagePrefix(severity) + message)
        }
    }
}
